kubernetes-with-otelcol-tracing/templates/collector.yaml

```
@@ -0,0 +1,100 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/name: {{ $.Values.baseName }}
+    app.kubernetes.io/component: opentelemetry-collector
+    service: buildbarn
+    team: {{ $.Values.team }}
+  name: {{ $.Values.baseName }}-opentelemetry-collector
+  namespace: {{ $.Release.Namespace }}
+spec:
+  replicas: {{ $.Values.otelcol.replicas }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ $.Values.baseName }}
+      app.kubernetes.io/component: opentelemetry-collector
+  template:
+    metadata:
+      annotations:
+        checksum/common.config: {{ include (print $.Template.BasePath "/config/collector.yaml.tpl") . | sha256sum }}
+        ad.datadoghq.com/otelcol.check_names: |
+          ["openmetrics"]
+        ad.datadoghq.com/otelcol.init_configs: |
+          [{}]
+        ad.datadoghq.com/otelcol.instances: |
+          [
+            {
+              "prometheus_url": "http://%%host%%:%%port_http%%/metrics",
+              "namespace": "buildbarn.otelcol",
+              "metrics": ["*"]
+            }
+          ]
+      labels:
```

```suggestion
      labels:
        app.kubernetes.io/name: {{ $.Values.baseName }}
```
This label is missing, while being present in the selector:
> Error: INSTALLATION FAILED: Deployment.apps "buildbarn-opentelemetry-collector" is invalid: spec.template.metadata.labels: Invalid value: map[string]string{"app.kubernetes.io/component":"opentelemetry-collector"}: `selector` does not match template `labels`